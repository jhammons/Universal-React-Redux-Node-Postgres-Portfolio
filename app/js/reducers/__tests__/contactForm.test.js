import contactFormReducer from '../contactForm';
import {
    SUBMIT_CONTACT_INVALID,
    SUBMIT_CONTACT_SENDING,
    SUBMIT_CONTACT_SUCCESS,
    SUBMIT_CONTACT_FAILED
} from '../../actions/contact';


describe('contactForm reducer', () => {
    const initialState = {
        readyState: SUBMIT_CONTACT_INVALID,
        result: null
    };

    test('initial state is correct', () => {
        const action = { type: 'test_action' };

        expect(contactFormReducer(undefined, action)).toEqual(initialState);
        expect(contactFormReducer(undefined, action)).toMatchSnapshot();
    });

    test('SUBMIT_CONTACT_SENDING', () => {
        const action = { type: SUBMIT_CONTACT_SENDING };
        const expectedState = {
            ...initialState,
            readyState: SUBMIT_CONTACT_SENDING,
        };

        expect(contactFormReducer(undefined, action)).toEqual(expectedState);
        expect(contactFormReducer(undefined, action)).toMatchSnapshot();
    });

    test('SUBMIT_CONTACT_FAILED', () => {
        const action = { type: SUBMIT_CONTACT_FAILED, error: 'big bad error' };
        const expectedState = {
            ...initialState,
            readyState: SUBMIT_CONTACT_FAILED,
            error: action.error
        };

        expect(contactFormReducer(undefined, action)).toEqual(expectedState);
        expect(contactFormReducer(undefined, action)).toMatchSnapshot();
    });

    test('SUBMIT_CONTACT_SUCCESS', () => {
        const action = { type: SUBMIT_CONTACT_SUCCESS, result: 'it worked!' };
        const expectedState = {
            ...initialState,
            readyState: SUBMIT_CONTACT_SUCCESS,
            result: action.result
        };

        expect(contactFormReducer(undefined, action)).toEqual(expectedState);
        expect(contactFormReducer(undefined, action)).toMatchSnapshot();
    });
});