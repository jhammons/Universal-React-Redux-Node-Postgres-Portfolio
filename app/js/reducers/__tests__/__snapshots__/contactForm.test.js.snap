// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contactForm reducer SUBMIT_CONTACT_FAILED 1`] = `
Object {
  "error": "big bad error",
  "readyState": "SUBMIT_CONTACT_FAILED",
  "result": null,
}
`;

exports[`contactForm reducer SUBMIT_CONTACT_RESET 1`] = `
Object {
  "readyState": "SUBMIT_CONTACT_INVALID",
  "result": null,
}
`;

exports[`contactForm reducer SUBMIT_CONTACT_SENDING 1`] = `
Object {
  "readyState": "SUBMIT_CONTACT_SENDING",
  "result": null,
}
`;

exports[`contactForm reducer SUBMIT_CONTACT_SUCCESS 1`] = `
Object {
  "readyState": "SUBMIT_CONTACT_SUCCESS",
  "result": "it worked!",
}
`;

exports[`contactForm reducer initial state is correct 1`] = `
Object {
  "readyState": "SUBMIT_CONTACT_INVALID",
  "result": null,
}
`;
